dpm/client/__init__.py

```diff
@@ -9,21 +9,30 @@
 class DpmException(Exception):
     pass
 
+
+class ConfigError(Exception):
+    """ The configuration passed to the client is malformed """
+    pass
+
+
 class Client(object):
 
-    def __init__(self, data_package_path='', config_path="~/.dpm/config", click=None):
+    def __init__(self, config, data_package_path='', click=None):
```

Comment:
And possibly even more useful command: search a datapackage on the registry server. That is given some text string perform full-text search on the indexed fields and return a list of matched packages. This command does not require a datapackage either.